### PR TITLE
Document vmctl overview and usage

### PR DESCRIPTION
**Pull request by stu-gott**
Documentation covering what vmctl is and how it can be used.

### CODE REVIEW

File: workloads/virtual-machines/vmctl.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+step behind. Any significant changes upstream would need to be implemented
+manually in KubeVirt.
+
+## Overview
+
+Vmctl is designed to address this delta by managing VirtualMachines from within
+a Pod. Vmctl will take an upstream VirtualMachine to act as a prototype and
+derive/spawn a new VirtualMachine based on it. This approach allows an
+arbitrary number of VirtualMachines that are nearly identical to be spawned
+using Kubernetes-native constructs. Each VirtualMachine is an exact copy of the
```
Comment (fabiand) [inline workloads/virtual-machines/vmctl.md]: I really like the paragraph until this point.

Maybe it's worth digging a little more into what this approach enables, something alogn the lines of:

> vmctl is running in a pod. The VM it dervied and spawned, will be runing alongside of it. Thus for every vmctl pod in the cluster, there should be a Vm running alongside of it. To be clear, vmctl _is not a VM_ instead it is controlling a VM close by.

File: workloads/virtual-machines/vmctl.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        downwardAPI:
+          items:
+          - path: "name"
+            fieldRef:
+              fieldPath: metadata.name
+```
+
+This example would look for a VirtualMachine in the `default` namespace named
+`testvm`, and instantiate a VirtualMachine on every node in the Kubernetes
+cluster.
```
Comment (fabiand) [inline workloads/virtual-machines/vmctl.md]: We should add a small section about how to integrate this with `Service`s.

I.e. a service can be created but needs to export the VM (not the vmctl pods) using a label/selector.

**Comment by fabiand** [comment]
@stu-gott maybe you should propose to move vmctl to the kubevirt org before getting this code merged.

File: workloads/virtual-machines/vmctl.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+      labels:
+        app: vmctl
+    spec:
+      containers:
+      - name: vmctl
+        image: quay.io/fabiand/vmctl
+        imagePullPolicy: IfNotPresent
+        args:
+        - "testvm"
+        volumeMounts:
```
[the remark this answers is not in the document]
Comment (fabiand) [reply workloads/virtual-machines/vmctl.md]: https://github.com/fabiand/vmctl/pull/13

**Comment by stu-gott** [comment]
Oops. Didn't realize we had labels. Added the "do not merge" label until we can address the source of the upstream repo.

**Comment by fabiand** [comment]
Much better.

Good that we have the preset.

**Pull request merged**